test/fedramp/fedramp_cluster_creation_test.go

```diff
@@ -0,0 +1,146 @@
+		}
+		err := provider.DeleteCluster(ctx, deleteOptions)
+		Expect(err).NotTo(HaveOccurred(), "Failed to delete ROSA cluster")
+	})
+
+	It("should successfully create a ROSA FedRamp Cluster", func(ctx context.Context) {
+		var err error
+
+		// Cluster setup variables
+		// TODO: Change this to a random name
+		clusterName = "osde2efr"
+		logger.Info(fmt.Sprintf("Cluster name set to: %s", clusterName))
+
+		clusterChannel = os.Getenv("CHANNEL")
+		if clusterChannel == "" {
+			clusterChannel = "stable"
+		}
+		logger.Info(fmt.Sprintf("Cluster channel set to: %s", clusterChannel))
+
+		// Get the cluster version
+		logger.Info("Fetching cluster version")
+		clusterVersion = os.Getenv("CLUSTER_VERSION")
+		if clusterVersion == "" {
+			clusterVersion, err = getClusterVersion(ctx, provider, clusterChannel)
+			Expect(err).NotTo(HaveOccurred())
+		}
+		logger.Info(fmt.Sprintf("Cluster version fetched: %s", clusterVersion))
+
+		privatelink = true
+		sts = true
+		logger.Info("Creating ROSA cluster with the following options:")
+		logger.Info(fmt.Sprintf("PrivateLink: %t, STS: %t, Version: %s", privatelink, sts, clusterVersion))
+
+		// Create ROSA cluster
+		clusterID, err = provider.CreateCluster(
+			ctx,
+			&rosa.CreateClusterOptions{
+				ClusterName:     clusterName,
+				ChannelGroup:    clusterChannel,
+				PrivateLink:     privatelink,
+				STS:             sts,
+				Version:         clusterVersion,
+				SkipHealthCheck: true, // TODO: Remove this once we have a way to access a privatelink cluster
+			},
+		)
+		Expect(err).NotTo(HaveOccurred(), "Failed to create ROSA cluster")
+		logger.Info(fmt.Sprintf("Cluster created successfully! ClusterID: %s", clusterID))
+	})
+})
+
+// getClusterVersion gets the default ROSA version for the given channel group
+// This is a hacky way to get the default ROSA version, but it works for now
+// TODO: Find a better way to get the default ROSA version
+// Latest rosa version is not available in fedramp rosa as they lag behind commercial.
+// Note the Provider isn't being used here, this function could be moved to osde2e-common after a review.
+func getClusterVersion(ctx context.Context, provider *rosa.Provider, clusterChannel string) (string, error) {
+	// Construct the command to list ROSA versions in JSON format
+	cmd := exec.CommandContext(ctx, "rosa", "list", "versions", "--channel-group", clusterChannel, "--output", "json")
+
+	var stdoutBuf, stderrBuf bytes.Buffer
+	cmd.Stdout = &stdoutBuf
+	cmd.Stderr = &stderrBuf
+
+	err := cmd.Run()
+	if err != nil {
+		return "", fmt.Errorf("failed to list ROSA versions: %w\n%s", err, stderrBuf.String())
+	}
+
+	type VersionInfo struct {
+		RawID   string `json:"raw_id"`
+		Default bool   `json:"default"`
+	}
+
+	var versions []VersionInfo
+	err = json.Unmarshal(stdoutBuf.Bytes(), &versions)
+	if err != nil {
+		return "", fmt.Errorf("failed to parse JSON output: %w", err)
+	}
+
+	if len(versions) == 0 {
+		return "", fmt.Errorf("no ROSA versions found")
```

Comment:
```suggestion
		return "", fmt.Errorf("no ROSA versions found in channel %s", clusterChannel)
```
linting will complain about using `Errorf` with no formatting directives/args. Same as the last return of this func